src/tiledimage.js

```diff
@@ -694,16 +694,11 @@ $.extend($.TiledImage.prototype, $.EventSource.prototype, /** @lends OpenSeadrag
         var isArray = function(obj) {
             return Object.prototype.toString.call(obj) === '[object Array]';
         };
-        var isXYPair = function(obj) {
-            return obj && isArray(obj) && obj.length === 2;
-        };
-        var convertXYObjectsToArrayIfNeeded = function(objs) {
+        var objectToSimpleXYObject = function(objs) {
             return objs.map(function(obj) {
                 try {
                     if (isXYObject(obj)) {
-                        return [obj.x, obj.y];
-                    } else if (isXYPair(obj)) {
-                        return obj;
+                        return { x: obj.x, y: obj.y };
```

Comment:
Removed the array support. 
Code will take whatever object with `x` and `y`, then convert it to a plain `x` and `y` object.
In drawTiles, they will be convert to `$.Point`, so I think it is not needed to convert them here.

Reply:
Thank you!

The remaining code here seems a little overkill (we could just copy the array that was given), but it does have several nice features:

* It validates that the data given is in the correct format and gives an error if not
* It copies the data so if the original is changed elsewhere, it doesn't affect us

...So I'm all for it :)